libgnucash/engine/qofinstance.cpp

```diff
@@ -114,7 +114,7 @@ typedef struct QofInstancePrivate
 #define GET_PRIVATE(o)  \
    (G_TYPE_INSTANCE_GET_PRIVATE ((o), QOF_TYPE_INSTANCE,  QofInstancePrivate))
 
-QOF_GOBJECT_GET_TYPE(QofInstance, qof_instance, G_TYPE_OBJECT, {});
+QOF_GOBJECT_GET_TYPE(QofInstance, qof_instance, G_TYPE_OBJECT, G_ADD_PRIVATE(QofInstance));
```

Comment:
I don't think that ```QOF_GOBJECT_FOO``` really bring much to the party and they're used in only 8 places so let's just use ```G_DEFINE_TYPE_WITH_PRIVATE``` here and in qofbook.cpp instead. Otherwise this is fine, proceed to the rest.